corehq/apps/app_manager/models.py

```diff
@@ -1573,22 +1573,29 @@ def custom_icon_form_and_text_by_language(self, lang):
             return custom_icon.form, custom_icon_text
         return None, None
 
-    def _set_media(self, media_attr, lang, media_path):
+    def _set_media(self, media_attr, lang, media_path, app=None):
         """
             Caller's responsibility to save doc.
             Currently only called from the view which saves after all Edits
         """
         assert media_attr in ('media_image', 'media_audio')
 
         media_dict = getattr(self, media_attr) or {}
+        old_value = media_dict.get(lang)
+        # remove the entry from app multimedia mappings if media is being removed now
+        # This does not remove the multimedia but just it's reference in mapping
+        if old_value and not media_path:
+            if app is None:
```

Comment:
@biyeun @orangejenny 

I had to add this check for `case_list` and `case_list_form` that don't have `get_app` or any way probably to find the app related. I ended up passing the app object from where its called to work this for now.

Is there a way around this by adding get_app to case_list and case_list_form which would be useful for other methods within this mixin that call get_app() as well but are probably not used for these two(case_list and case_list_form) and hence causing no issues, but would be nice to add for consistency.
or this the only possible approach?

Reply:
I agree it'd be better to make `get_app` a requirement for implementing `NavMenuItemMediaMixin` and then add `get_app` to the case list and case list form. Can you add either app id or module id to CaseList and CaseListForm, set it in ModuleBase.wrap, and then add get_app methods to CaseList and CaseListForm? (I'm guessing a little bit here, but think I've seen similar code in HQ to set data that isn't stored in couch but is available at the time the object is wrapped.)

Reply:
Cool, gave that a shot and it needed a couple of more changes. Does not look like the best of ways to do this probably. let me know how that looks and if it can be implemented in a better manner.